sources/nvidia-migmanager/src/main.rs

```diff
@@ -0,0 +1,688 @@
+            .enumerate()
+            .filter(|(_, gpu)| gpu.model == NvidiaGpu::Other)
+            .map(|(index, _)| index)
+            .collect();
+        let known_gpus: Vec<NvidiaGpuModel> = vec![
+            NvidiaGpuModel::try_from("a100.40gb").unwrap(),
+            NvidiaGpuModel::try_from("a100.80gb").unwrap(),
+            NvidiaGpuModel::try_from("h100.80gb").unwrap(),
+            NvidiaGpuModel::try_from("h200.141gb").unwrap(),
+        ]
+        .into_iter()
+        .collect();
+
+        // The GPU in the current instance is not one of the known GPUs. We attempt using a profile that doesn't belong to one of the known GPUs.
+        for (gpu, mig_profile) in &mig_settings.profile {
+            if !known_gpus.contains(gpu) {
+                //If the GPU is unknown, we want the exact MIG Profile and not the number of slices.
+                ensure!(mig_profile.len() > 1, error::MigProfileSnafu {});
+
+                let gpu_regex = Regex::new(GPU_MODEL_REGEX).unwrap();
+                let profile_regex = Regex::new(MIG_PROFILE_REGEX).unwrap();
+
+                let gpu_ram: usize = gpu_regex
+                    .captures(gpu)
+                    .map(|captures| captures[1].parse().unwrap_or(0))
+                    .unwrap();
+                let (compute_slices, slice_ram) = profile_regex
+                    .captures(mig_profile)
+                    .map(|captures| {
+                        (
+                            captures[1].parse().unwrap_or(0),
+                            captures[2].parse().unwrap_or(0),
+                        )
+                    })
+                    .unwrap();
+
+                let num_slices: usize = min(gpu_ram / slice_ram, 7 / compute_slices);
+                let profile_string = std::iter::repeat(mig_profile.as_ref())
+                    .take(num_slices)
+                    .collect::<Vec<_>>()
+                    .join(",");
+
+                set_mig_profile(other_indices.clone(), profile_string.as_str())?;
+            }
+        }
+    } else {
+        let (is_ampere_gpu_present, has_enabled_mig) =
+            gpu_info
+                .iter()
+                .fold((false, false), |(ampere, enabled), gpu| {
+                    let is_ampere =
+                        gpu.model == NvidiaGpu::A100_40GB || gpu.model == NvidiaGpu::A100_80GB;
+                    let is_enabled = gpu.state == MigState::Enabled;
+                    (ampere || is_ampere, enabled || is_enabled)
+                });
+
+        if has_enabled_mig {
+            // Disable MIG for the GPU
+            set_mig_mode(false)?;
+
+            // If GPU is A100 create marker file for reboot to reconcile
+            // for the gpu reset and move from transitional state to disabled
+            if is_ampere_gpu_present {
+                info!("Rebooting to apply MIG Settings.");
+                std::fs::write("/run/nvidia-migmanager/reboot-required", "Disabling MIG").unwrap();
```

Comment:
Please take another pass through and convert fallible `unwrap()` calls like this into snafu errors.

For `Regex::new` and other places that are nominally fallible but given known inputs at compile time, you can use either `unwrap` or `expect` - the latter lets you provide an error message which can also help explain why it should never happen.